### PR TITLE
sprufication of templates

### PR DESCRIPTION
**Pull request by longnguyen11288**
Switch to using spruce instead of spiff, can be downloaded at https://github.com/geofffranks/spruce/releases

### CODE REVIEW

File: lib/bosh/gen/generators/new_release_generator/templates/templates/make_manifest.tt [hunk 1]
```diff
@@ -59,7 +59,7 @@ perl -pi -e "s/PLACEHOLDER-DIRECTOR-UUID/$DIRECTOR_UUID/g" $tmpdir/stub-with-uui
 perl -pi -e "s/NAME/$NAME/g" $tmpdir/stub-with-uuid.yml
 perl -pi -e "s/STEMCELL/$STEMCELL/g" $tmpdir/stub-with-uuid.yml
 
-spiff merge \
+spruce merge --prune meta \
```
Comment (drnic) [inline lib/bosh/gen/generators/new_release_generator/templates/templates/make_manifest.tt]: Can you do me & others a favour and do a `if [[ "$(which spruce)X" == "X" ]]` to test if spruce is installed; and if not, show them where/how to get it?

Comment (longnguyen11288) [inline lib/bosh/gen/generators/new_release_generator/templates/templates/make_manifest.tt]: done

File: lib/bosh/gen/generators/new_release_generator/templates/templates/make_manifest.tt [hunk 2]
```diff
@@ -59,7 +59,11 @@ perl -pi -e "s/PLACEHOLDER-DIRECTOR-UUID/$DIRECTOR_UUID/g" $tmpdir/stub-with-uui
 perl -pi -e "s/NAME/$NAME/g" $tmpdir/stub-with-uuid.yml
 perl -pi -e "s/STEMCELL/$STEMCELL/g" $tmpdir/stub-with-uuid.yml
 
-spiff merge \
+if ! [ -x "$(command -v spruce)" ]; then
```
Comment (drnic) [inline lib/bosh/gen/generators/new_release_generator/templates/templates/make_manifest.tt]: ok, thanks; but move this validation to the top of the script, before the expensive `bosh` calls, so we get fast failure

I tested the PR; worked. Merge when you're ready.

**Comment by drnic** [comment]
Since this introduces spruce to many people - perhaps document/intro it in both bosh-gen's README _and_ the generated README.

**Pull request merged**